pkg/apis/core/v1/resource_genclient.go

```diff
@@ -91,7 +90,7 @@ func (rwt ResourceWithTag) EndpointURL(ctx context.Context) (*url.URL, error) {
 		return nil, fmt.Errorf("%w: ResourceWithTag only support Create and Destroy operations", api.ErrOperationNotSupported)
 	}
 
-	return url.Parse(fmt.Sprintf("/api/core/v1/resource.json/%v/tags/%v", rwt.Identifier, rwt.Tag))
+	return url.Parse(fmt.Sprintf("/api/core/v1/resource.json/%v/tags/%v", rwt.ResourceIdentifier, rwt.Tag))
```

Comment:
:warning: This would break *all* existing usages for this resource!

The ticket you've created says:

> When using the mock, this causes a resource object to be replaced in the mock data with the ResourceWithTag object because they have the same identifier.

It's clear that this is a problem inside the implementation of the mock, not of the actual API. There are two solutions to this problem:

1. Either provide a way that sets the `ResourceIdentifier` in a backwards compatible way for existing users *or*
2. fix the wrong behavior in the mock itself.

My preference would be option 2, but I'm fine with option 1 as well.

Reply:
We agreed on adjusting the code so it is backwards compatible, using `Identifier` when no `ResourceIdentifier` is available.

Reply:
Thanks for that feedback, I additionally found an issue with the tests and fixed it as well.